ee/clickhouse/models/property.py

```diff
@@ -32,12 +33,100 @@
     SELECT_PERSON_PROP_VALUES_SQL,
     SELECT_PERSON_PROP_VALUES_SQL_WITH_FILTER,
 )
+from posthog.constants import PropertyOperatorType
 from posthog.models.cohort import Cohort
 from posthog.models.event import Selector
-from posthog.models.property import NEGATED_OPERATORS, OperatorType, Property, PropertyIdentifier, PropertyName
+from posthog.models.property import (
+    NEGATED_OPERATORS,
+    OperatorType,
+    Property,
+    PropertyGroup,
+    PropertyIdentifier,
+    PropertyName,
+)
 from posthog.models.team import Team
 from posthog.utils import is_valid_regex, relative_date_parse
 
+# Property Groups Example:
+# {type: 'AND', groups: [
+#     {type: 'OR', groups: [A, B, C]},
+#     {type: 'OR', groups: [D, E, F]},
+# ]}
+
+# Example:
+# {type: 'AND', groups: [
+#     A, B, C, D
+# ]}
+
+# Property json is of the form:
+# { type: 'AND | OR', groups: List[Property] }
+# which is parsed and sent to this function ->
+
+
+def parse_prop_grouped_clauses(
+    property_group: PropertyGroup,
+    prepend: str = "global",
+    table_name: str = "",
+    allow_denormalized_props: bool = True,
+    has_person_id_joined: bool = True,
+    person_properties_mode: PersonPropertiesMode = PersonPropertiesMode.USING_SUBQUERY,
+    person_id_joined_alias: str = "person_id",
+    group_properties_joined: bool = True,
+    _top_level: bool = True,
+) -> Tuple[str, Dict]:
+
+    if len(property_group.groups) == 0:
+        return "", {}
+
+    if isinstance(property_group.groups[0], PropertyGroup):
```

Comment:
It's not obvious why this is correct - why can you only send one PropertyGroup at a time?

This isn't verified anywhere -> leads to unexpected behavior. Note that these APIs are not completely private and will be more documented and public long-term

At a minimum, an assert should be here.